FileEmulationFramework.Lib/Memory/MemoryManagerStream.cs

```diff
@@ -99,7 +99,11 @@ protected override void Dispose(bool disposing)
     public override int Read(Span<byte> buffer)
     {
         EnsureAccessorNoAlloc();
+
         var count = buffer.Length;
+        
+        if (Length - Position < count)
```

Comment:
Shouldn't this code only be present in the 'else' branch below?
When `count <= _bytesAvailable`, it shouldn't be possible that BytesRemaining (`Length - Position`) is lesser than count.

Reply:
Yes, I think you're right. I've fixed this in df85ae3